org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/preferences/LifecycleMappingsViewer.java

```diff
@@ -444,52 +475,56 @@ String getSourcelabel(MojoExecutionKey execution, List<IPluginExecutionMetadata>
     LinkedHashSet<String> sources = new LinkedHashSet<>();
     if(mappings != null && !mappings.isEmpty()) {
       for(IPluginExecutionMetadata mapping : mappings) {
-        LifecycleMappingMetadataSource metadata = ((PluginExecutionMetadata) mapping).getSource();
-        if(metadata != null) {
-          Object source = metadata.getSource();
-          if(source instanceof String s) {
-            sources.add(s);
-          } else if(source instanceof Artifact artifact) {
-            sources.add(getSourceLabel(artifact, detailed));
-          } else if(source instanceof MavenProject mavenProject) {
-            sources.add(getSourceLabel(mavenProject, detailed));
-          } else if(source instanceof Bundle bundle) {
-            sources.add(getSourceLabel(bundle, detailed));
+        if(mapping instanceof PluginExecutionMetadata metadata) {
+          LifecycleMappingMetadataSource metadataSource = metadata.getSource();
+          if(metadataSource == null) {
+            sources.add(UNKNOWN);
           } else {
-            sources.add("unknown"); //$NON-NLS-1$
+            Object source = metadataSource.getSource();
+            if(source instanceof String s) {
+              sources.add(s);
+            } else if(source instanceof Artifact artifact) {
+              sources.add(getSourceLabel(artifact, detailed));
+            } else if(source instanceof MavenProject mavenProject) {
+              sources.add(getSourceLabel(mavenProject, detailed));
+            } else if(source instanceof Bundle bundle) {
+              sources.add(getSourceLabel(bundle, detailed));
+            } else {
+              sources.add(UNKNOWN); //$NON-NLS-1$
+            }
           }
         } else if(mapping instanceof DefaultPluginExecutionMetadata) {
-          sources.add("default"); //$NON-NLS-1$
+          sources.add(DEFAULT); //$NON-NLS-1$
         } else {
```

Comment:
this one as well, so we have a chain like 

```
if(mapping instanceof PluginExecutionMetadata metadata) {
 ...
} else if(mapping instanceof DefaultPluginExecutionMetadata) {
 ...
} else {
  sources.add(UNKNOWN):
}
```

Reply:
See above, this is exactly how it is.